### DISCUSSION TITLE
How can I prevent 'scrolling to the top of the page behavior' when clicking on clr-dg-action-overflow in a datagrid that overflows its container on the y axis?

### DISCUSSION THREAD

**Discussion by ghost** [vmware/clarity#6438] [Q&A]
## Describe the bug

I have a page that I use clr-datagrid. I want to make some actions and for this reason I use clr-dg-action-overflow tag. However when I click on the clr-dg-action-overflow, although the menu opens, the whole page scrolls to the top.

## How to reproduce

The error can be reproduced in  [Clarity Project v5](https://stackblitz.com/edit/clarity-light-theme-v5-6h8avy).

1.  Go to The last item (name98)
2.  Click on 3dots (Available Items)
3.  The page scrolls to the top
4.  The menu appears near name 98

## Expected behavior

The page shouldn't scroll to the top

## Versions

**Clarity project:**

- [ 5.4.0] CDS core
- [5.4.- ] Clarity Angular/UI

**Clarity version:**

- [ ] v3.x
- [ ] v4.x
- [X] v5.x

**Framework:**

- [ X] Angular
- [ ] React
- [ ] Vue
- [ ] Other:

**Framework version:**
_ie: Angular 10,11,12_

**Device:**

- Type: HP
- OS: Windows 10
- Browser Chrome
- Version 95.0.4638.54

**Comment by bdryanovski** [comment]
I could confirm the issue, don't have an idea why this is happening - it seems that we need to use `preventDefault` for the click but I'm not sure that is that only.

**Comment by hippee-lee** [comment]
The example does scroll but the behavior is related to some early css/layout decisions we made back in 2016. The correct way to have a datagrid with rows that overflow its container is to use it in the `content-area` class of the Clarity application layout. Clarity (angular) application layout is described [here](https://angular.clarity.design/documentation/app-layout). 

I forked the stackblitz and added the correct Clarity (angular) application layout. If using the application layout as documented is not an option you will need to recreate something similar with flexbox or grid so that the  container for the datagrid has an auto overflow-y and the body doesn't have any content overflow. 

This has come up before so instead of closing the issue I'll convert this to a discussion so that future folks can find it.

**Reply by ghost** [reply]
Could you please send me the forked stackblitz link?

**Reply by hippee-lee** [reply]
Sure, Sorry, I thought I added the link to the markdown. [Here](https://stackblitz.com/edit/clarity-light-theme-v5-xhno3g?file=src/app/app.component.html) it is.
